modules/python/python-fetcher.c

```diff
@@ -384,6 +386,30 @@ _py_parse_options_new(PythonFetcherDriver *self, MsgFormatOptions *parse_options
   return py_parse_options;
 }
 
+static gboolean
+_py_init_ack_tracker_factory(PythonFetcherDriver *self)
+{
+  PyObject *py_ack_tracker_factory = _py_get_attr_or_null(self->py.instance, "ack_tracker");
+
+  if (!py_ack_tracker_factory)
+    return TRUE;
+
+  if (!py_is_ack_tracker_factory(py_ack_tracker_factory))
+    {
+      msg_error("Python source attribute ack_tracker needs to be an AckTracker subtype",
+                evt_tag_str("driver", self->super.super.super.super.id),
+                evt_tag_str("class", self->class));
+      return FALSE;
+    }
+
+  self->py.ack_tracker_factory = (PyAckTrackerFactory *) py_ack_tracker_factory;
+
+  AckTrackerFactory *ack_tracker_factory = self->py.ack_tracker_factory->ack_tracker_factory;
+  self->super.super.worker_options.ack_tracker_factory = ack_tracker_factory;
```

Comment:
`ack_tracker_factory_unref` is called in `log_threaded_source_worker_options_destroy`, so I think `ack_tracker_factory` should be referenced.

Reply:
Fixed, thank you.